fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/cursors/ConcatCursorTest.java

```diff
@@ -0,0 +1,438 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.cursors;
+
+import com.apple.foundationdb.KeyValue;
+import com.apple.foundationdb.record.ExecuteProperties;
+import com.apple.foundationdb.record.RecordCursor;
+import com.apple.foundationdb.record.RecordCursorResult;
+import com.apple.foundationdb.record.RecordCursorVisitor;
+import com.apple.foundationdb.record.ScanProperties;
+import com.apple.foundationdb.record.TupleRange;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabase;
+import com.apple.foundationdb.record.provider.foundationdb.FDBDatabaseFactory;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordContext;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStore;
+import com.apple.foundationdb.record.provider.foundationdb.FDBRecordStoreKeyspace;
+import com.apple.foundationdb.record.provider.foundationdb.KeyValueCursor;
+import com.apple.foundationdb.record.provider.foundationdb.SizeStatisticsCollector;
+import com.apple.foundationdb.record.provider.foundationdb.TestKeySpace;
+import com.apple.foundationdb.record.provider.foundationdb.cursors.TriFunction;
+import com.apple.foundationdb.record.provider.foundationdb.keyspace.KeySpacePath;
+import com.apple.foundationdb.subspace.Subspace;
+import com.apple.foundationdb.tuple.Tuple;
+import com.apple.test.Tags;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * Tests for {@link ConcatCursor}.
+ */
+@Tag(Tags.RequiresFDB)
+public class ConcatCursorTest {
+
+    FDBDatabase fdb;
+    FDBRecordContext context;
+    private Subspace subspace;
+    ExecuteProperties ep;
+
+    @BeforeEach
+    public void setup() throws Exception {
+        fdb = FDBDatabaseFactory.instance().getDatabase();
+        context = fdb.openContext();
+    }
+
+    @AfterEach
+    public void teardown() throws Exception {
+        context.close();
+        fdb.close();
+    }
+
+    @Test
+    public void concatCursorBasicTests() {
+
+        concatListCursorTests(ScanProperties.FORWARD_SCAN);
+        concatListCursorTests(ScanProperties.REVERSE_SCAN);
```

Comment:
These tests are kind of their own thing.  Maybe it would be better to have:

    @Test 
    public void simpleForwardScan() {
        concatListCursorTests(ScanProperties.FORWARD_SCAN);
    }

    @Test 
    public void simpleReverseScan() {
        concatListCursorTests(ScanProperties.REVERSE_SCAN);
    }

Reply:
done